hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/codec/BlockInfoTranslator.java

```diff
@@ -43,7 +43,7 @@ public static BlockInfo blockInfoFromMerkleNetworkContext(
             throws IOException {
         requireNonNull(merkleNetworkContext);
         final var blockInfoBuilder = new BlockInfo.Builder()
-                .lastBlockNumber(merkleNetworkContext.getAlignmentBlockNo())
+                .lastBlockNumber(merkleNetworkContext.getAlignmentBlockNo() - 1)
```

Comment:
This arbitrary-seeming `-1` will possibly puzzle the future reader.  Perhaps a comment to indicate what it's for?  (I mean, I know its to fix a diff testing issue, but why `-1` and not `+0`?)

Reply:
Added comment!